File: 392-IsSubsequence/392-IsSubsequence.py
# Last updated: 8/1/2025, 10:37:25 PM
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        find=0
        for i in s:
            if i in t:
                find+=1
                indx=t.index(i)
                t=t[indx+1::]
        if find==len(s):
            return True
        else:
            return False            

File: 392-IsSubsequence/392-IsSubsequence.java
// Last updated: 11/7/2025, 11:33:11 PM
class Solution {
    public boolean isSubsequence(String s, String t) {
        int l=0;
        int r=0;
        while(l<s.length() && r<t.length()){
            char ch1=s.charAt(l);
            char ch2=t.charAt(r);
            if(ch1==ch2){
                l++;
            }
            
            r++;
        }
        return s.length()==l;
    }
}